src/utils/hooks/useConfirmationModal.tsx

```diff
@@ -36,13 +38,19 @@ const useConfirmationModal = ({
     onConfirm: onConfirm ? onConfirm : () => console.log('Clicked OK'),
     onCancel: onCancel ? onCancel : () => {},
     showCancel,
+    awaitAction,
```

Comment:
Added this extra `awaitAction` prop so we have the option to close the modal *before* the confirm action runs. Otherwise, since the snapshot operation loads a modal of its own, it looks wrong with a transparent warning modal persisting under the snapshot modal. (But normally you would want this on, so it's the default)